x/shared/helpers/service_test.go

```diff
@@ -28,6 +32,78 @@ func TestIsValidServiceId(t *testing.T) {
 	}
 }
 
+func TestIsValidServiceName(t *testing.T) {
+	tests := []struct {
+		input    string
+		expected bool
+	}{
+		{"ValidName-1", true},
+		{"Valid Name_1", true},
+		{"valid name with spaces", true},
+		{"invalid@name", false}, // contains invalid character '@'
+		{"Valid.Name", false},   // contains invalid character '.'
+		{"", true},              // empty string is valid for ServiceName
+		{"validnamebuttoolongvalidnamebuttoolongvalidnamebuttoolong", false}, // exceeds maxServiceIdName length
+	}
+
+	for _, test := range tests {
+		t.Run(test.input, func(t *testing.T) {
+			result := IsValidServiceName(test.input)
+			if result != test.expected {
+				t.Errorf("For input %s, expected %v but got %v", test.input, test.expected, result)
+			}
+		})
+	}
+}
+
+func TestIsValidService(t *testing.T) {
```

Comment:
I think this is [duplicated in #150](https://github.com/pokt-network/poktroll/pull/150/files/2b95bc48e7eb8651701937a2fbf4231762e8e96b#diff-787693a5c2759a1d7cfa40155745aadc2ff109cb49bfe446952aea388adeea67R9).

Reply:
Good point. Let's merge #150 in first and then I'll resolve the conflicts.